tests/test_misc_table.py

```diff
@@ -606,6 +606,73 @@ def test_dtype_multiindex_single_level(
     assert db_new['misc'].index.dtype == expected_pandas_dtype
 
 
+def test_drop_and_pick_index():
+
+    table_id = 'misc'
+    index = pytest.DB[table_id].index[:5]
+    df_pick = pytest.DB[table_id].pick_index(index).get()
+    index = pytest.DB[table_id].index[5:]
+    df_drop = pytest.DB[table_id].drop_index(index).get()
+    pd.testing.assert_frame_equal(df_pick, df_drop)
+
+    index = pytest.DB['segments'].index[:5]
+    with pytest.raises(
+        ValueError,
+        match='Cannot drop',
+    ):
+        pytest.DB[table_id].drop_index(index).get()
+    with pytest.raises(
+        ValueError,
+        match='Cannot pick',
+    ):
+        pytest.DB[table_id].pick_index(index).get()
+
+
+def test_extend_index():
+
+    db = audformat.testing.create_db(minimal=True)
+    db.schemes['scheme'] = audformat.Scheme()
+
+    # empty and invalid
+
+    db['misc'] = audformat.MiscTable(pd.Index([], name='idx'))
+    db['misc'].extend_index(pd.Index([], name='idx'))
+    assert db['misc'].get().empty
+    with pytest.raises(
+        ValueError,
+        match='Cannot extend',
+    ):
+        db['misc'].extend_index(pd.Index([], name='other'))
+
+    db.drop_tables('misc')
+
+    # not empty
+
+    db['misc'] = audformat.MiscTable(pd.Index([], name='idx'))
+    db['misc']['columns'] = audformat.Column(scheme_id='scheme')
+    db['misc'].extend_index(
+        pd.Index(['1', '2'], name='idx'),
+        fill_values='a',
+        inplace=True,
+    )
+    np.testing.assert_equal(
+        db['misc']['columns'].get().values,
+        np.array(['a', 'a']),
+    )
+    index = pd.Index(['1', '3'], name='idx')
```

Comment:
What happens when we exten by `pd.MultiIndex.from_arrays([['1', '3']], anmes=['idx'])` instead?
It might result to an error as we convert a mixture of `Index` and `MultiIndex` always to `MultiIndex` in our helper functions.

Reply:
And another good one. Turns out that it indeed messes up the index:

```python
index = pd.MultiIndex.from_arrays([['1', '3']], names=['idx'])
db['misc'].extend_index(
    index,
    fill_values='b',
    inplace=True,
)
db['misc'].get()
```
```
     columns
1          a
2          a
(1,)       b
(3,)       b
```

To avoid such side-effect I suggest we do not allow to combine `Index` with a single-level `MultIndex` in `union()`

Reply:
I created https://github.com/audeering/audformat/issues/227

Reply:
I now switched to `utils.union()` inside `extend_index()` and added a test where we extend with a `pd.MultiIndex()`, but we need to solve #227 first.

Reply:
Ok, we are almost there. I added not tests where I drop, extend and pick also with a `pd.MultiIndex`. Two tests are failing because in `drop_index()` we are calling `self.df.index.difference(index)`, which will return a `pd.MultiIndex` instead of `pd.Index`. To solve this issue we have to add another utility function `utils.difference()` first. Will do this in a separate PR.

Reply:
Ok, I now switched to `utils.symmetric_difference() in `drop_index()` and tests are passing again.